### DISCUSSION TITLE
Is there a way to check if there is an active websocket connection?

### DISCUSSION THREAD

**Discussion by neliel123** [miguelgrinberg/microdot#183] [Q&A]
I noticed something strange in my application while working with Asynchronous WebSocket.
I am putting my messages in my message queue and sending it through WebSocket like the below code.
```
@app.route('/ws')
@with_websocket
async def read_queue(request, ws):
    while True:
        message = await message_queue.get()
        print(f"Message from queue : {message}")
        if message is not None:
            await ws.send(ujson.dumps(message))
```

If I continually interact with my MicroDot web application then I am able to receive the WebSocket messages.

Now, if I go out for like minutes or hours and leave my phone.  When I go back and refresh my phone browser, my web browser continually updates my interfaces as if it were buffered by the server,

When I look at the server logs, I do see the printout of the messages.

Is there a way to check in the MicroDot server if a client is connected or not?  If none, then just disregard the message.

**Comment by miguelgrinberg** [comment]
Maintaining long lived connections from mobile phones is tricky, because the phone may be constantly switching networks, or between wi-fi and cellular. The most reliable way to ensure the client and the server are still communicated is for them to exchange ping/pong packets at regular intervals. For example, the server can send a message with the word `"ping"`, and if it does not receive a reply with the word `"pong"` within 5 seconds it assumes the client is gone and ends the connection. Likewise, if the client does not receive a `"ping"` message approximately every 30 seconds it assumes the server is gone and attempts to reconnect. This is the technique used by protocols built on top of WebSocket such as Socket.IO.
